### PR TITLE
feat(gazelle): simplify gazelle_python.yaml

### PR DESCRIPTION
**Pull request by hunshcn**
Simplify and make gazelle_python.yaml have only top level package name.

It would work well in cases to reduce merge conflicts.

Resolve https://github.com/bazelbuild/rules_python/issues/1890

This is a compatible change and seems to have no side effects, so no options are provided.

### CODE REVIEW

File: gazelle/pythonconfig/pythonconfig.go [hunk 1]
```diff
@@ -494,9 +502,9 @@ func (c *Config) FormatThirdPartyDependency(repositoryName string, distributionN
 		normConventionalDistributionName = strings.Trim(normConventionalDistributionName, "_")
 	case Pep503LabelNormalizationType:
 		// See https://packaging.python.org/en/latest/specifications/name-normalization/#name-format
-		normConventionalDistributionName = strings.ToLower(conventionalDistributionName) // ... "should be lowercased"
+		normConventionalDistributionName = strings.ToLower(conventionalDistributionName)                                        // ... "should be lowercased"
 		normConventionalDistributionName = regexp.MustCompile(`[-_.]+`).ReplaceAllString(normConventionalDistributionName, "-") // ... "all runs of the characters ., -, or _ replaced with a single -"
-		normConventionalDistributionName = strings.Trim(normConventionalDistributionName, "-") // ... "must start and end with a letter or number"
+		normConventionalDistributionName = strings.Trim(normConventionalDistributionName, "-")                                  // ... "must start and end with a letter or number"
```
Comment (aignas) [inline gazelle/pythonconfig/pythonconfig.go:507]: nit, the addition of the whitespace is weird.

**Review by aignas — COMMENTED** [review]
@linzhp, @dougthor42, you have bigger projects using `gazelle` so it would be great to hear if this has any negative impacts. @alexeagle, since you have clients using `gazelle` maybe you have time to test this change out with bigger projects to let us know how this would work in cases where there is a lot of traffic and changes to what internal files packages have would cause merge conflicts?

File: gazelle/manifest/testdata/gazelle_python.yaml [hunk 2]
```diff
@@ -1,13 +1,5 @@
 manifest:
   modules_mapping:
     arrow: arrow
```
Comment (aignas) [inline gazelle/manifest/testdata/gazelle_python.yaml:3]: Could you please add a test where we have a different distribution providing a namespace package? Something like following would be good to have. We should check that importing arrow.plugin correctly adds arrow_plugin to the deps.
```suggestion
    arrow: arrow
    arrow.plugin: arrow_plugin
```

Comment (hunshcn) [reply gazelle/manifest/testdata/gazelle_python.yaml:3]: I believe this case will be handled correctly because it is covered in example/build_file_generation.

But it does make sense to add the test. I will add it at night.

Comment (hunshcn) [reply gazelle/manifest/testdata/gazelle_python.yaml:3]: added

**Comment by dougthor42** [comment]
Just FYI this is still on my radar. I expect to have some time in the next 48hrs.

**Comment by aignas** [comment]
Yeah, I'll wait to give enough time for people to look into this.

**Comment by dougthor42** [comment]
**Note:** I had to cherry-pick cca1d4efe49733217e7df30a84a172ad3d6e4ae1 (#2058) to get this to work, but that's expected.

`gazelle_python.yaml` is now just **2%** :exclamation: of the original size: It went from 28737 lines down to 549 lines. The "1 insertion" was simply the file hash changing.

```
 gazelle_python.yaml | 28190 +----------------------------------------------------------------------------------
 1 file changed, 1 insertion(+), 28189 deletions(-)
```


We have 505 BUILD.bazel files. I tried randomly deleting some of them and having gazelle remake them. No change[^1] :+1: 

```console
$ find src/ -type f -name BUILD.bazel -print0 | xargs -0 -I {} bash -c '[ $[ $RANDOM % 100 ] == 0 ] && echo deleting {} && rm {}'
list of 6 files
$ bazel run //:gazelle
$ git status
On branch u/dthor/bazel-simplify-gazelle-check
nothing to commit, working tree clean
```

I did the "randomly delete" again a couple more times. All good!

I also ran some `bazel test`, but I'd expect this to have zero impact on them. They ran just fine.

Overall I think this is :100: . Very nice! :tada: :partying_face: 

[^1]: Aside from expected changes that gazelle doesn't deal with, like `data` attribute or when my random delete removed a hand-made BUILD file.

**Comment by linzhp** [comment]
Shorten gazelle_python.yaml greatly. I didn't run into issues with some quick test

File: gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml [hunk 3]
```diff
@@ -0,0 +1,1678 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+manifest:
+  modules_mapping:
+    cachetools: cachetools
+    cachetools.__init__: cachetools
+    cachetools.func: cachetools
+    cachetools.keys: cachetools
+    certifi: certifi
```
Comment (aignas) [inline gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml:21]: This looks like copy paste from somewhere else. The codebase no longer generates these long files. Could you please fix this?

**Review by aignas — APPROVED** [review]
Minor nit, but otherwise looks good. I'll wait for a few more days for someone form Aspect to chime in.

File: gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml (hunk 3 above)
Comment (hunshcn) [reply gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml:21]: According to https://github.com/bazelbuild/rules_python/blob/36ed9e029c275256b520b1b5d0d3329664441ae0/gazelle/python/testdata/with_third_party_requirements_from_imports/README.md. The gazelle_python.yaml is generated from https://github.com/sramirezmartin/gazelle-toy-example/blob/main/subdir1/requirements_lock.txt.

Do you think I should add this file to git?

Comment (aignas) [reply gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml:21]: Just cleaning up the `gazelle_python.yaml` would be sufficient here. This is test data. Having it generated may be too much upkeep.

Comment (hunshcn) [reply gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml]: test `with_third_party_requirements_from_imports_legacy` has been removed. It was meant to be a compatibility test for the original format. But the guys' manual testing has ensured its usability.

**Comment by alexeagle** [comment]
I'm sorry, I have no time the rest of this month - maybe @jbedard has an idea since he worked on python gazelle recently

File: gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml (hunk 3 above)
Comment (aignas) [reply gazelle/python/testdata/with_third_party_requirements_from_imports_legacy/gazelle_python.yaml]: Removing sounds good.
